lib/open_mic.rb

```diff
@@ -0,0 +1,33 @@
+class OpenMic
+  attr_reader :location, :date, :performers
+
+  def initialize(open_mic_info)
+    @location = open_mic_info[:location]
+    @date = open_mic_info[:date]
+    @performers = []
+  end
+
+  def welcome(performer_name)
+    @performers.push(performer_name)
+  end
+
+  def repeated_jokes?
```

Comment:
The logic here looks great, but an easier way of checking if there are duplicates is to create an array of all known jokes and compare its length to a unique list of known jokes:

```
all_jokes = @performers.flat_map do |performer|
  performer.jokes
end

all_jokes.uniq.length != all_jokes.length
```